main/src/com/google/refine/expr/functions/strings/Contains.java

```diff
@@ -47,8 +49,14 @@ public Object call(Properties bindings, Object[] args) {
         if (args.length == 2) {
             Object s1 = args[0];
             Object s2 = args[1];
-            if (s1 != null && s2 != null && s1 instanceof String && s2 instanceof String) {
-                return ((String) s1).indexOf((String) s2) > -1;
+            if (s1 != null && s2 != null && s1 instanceof String && (s2 instanceof String || s2 instanceof Pattern)) {
```

Comment:
Do a better distinction between String and Pattern, as in the [replace() method](https://github.com/OpenRefine/OpenRefine/blob/master/main/src/com/google/refine/expr/functions/strings/Replace.java) ?